metrics/formulaCalculator.go

```diff
@@ -0,0 +1,77 @@
+package metrics
+
+import (
+	"math"
+)
+
+type Factor struct {
+	Weight       float64
+	Value        float64
+	AllOrNothing bool
+}
+
+func ComputeNetScore(fs []Factor) float64 {
+	var sum float64
+
+	for _, f := range fs {
+		if f.AllOrNothing {
+			if f.Value == 0 {
+				sum = 0
+				break
+			} else {
+				continue
+			}
+		}
+
+		sum += f.Value * f.Weight
+	}
+
+	return sum
+}
+
+func ComputeRampTime(found int, total int) float64 {
+	// Compute Ramp-up time based on number of phrases found in README
+	if total == 0 {
+		return 0
+	}
+	return float64(found) / float64(total)
+}
+
+func ComputeCorrectness(clones int, views int, commits int) float64 {
+	// Compute the correctness score based on number of repository
+	// clones, page views, and number of total commits
+	var cs, vs, ms float64
+	cs = 0.117 * (1 - math.Exp(-0.001*float64(clones)))   // 2k clones for 80%
+	vs = 0.550 * (1 - math.Exp(-0.00002*float64(clones))) // 100k views for 86%
+	ms = 0.333 * (1 - math.Exp(-0.0005*float64(clones)))  // 6000 commits for 90% of this
```

Comment:
i dont understand your comments for this whatsoever. I do understand how these things add up

Reply:
Yeah mb, I'll update the comment to make it more clear. Here's the comment I will add in

> Correctness is determined by a sum three factors: clones, views, and commits. Each of these are calculated using a an exponential decay function to ensure that the domain is from -inf to inf, while the range is still between 0 and the weight (0.117, 0.550, or 0.333).  As a rough benchmark, I determined the quantity of each metric to reach a certain output value.
> 
> Example:
> For clones, the weight is 0.117. To get 80% of that weight, we need the repository to have 2000 clones. The result is cs = 0.117 * 0.8 = 0.0936

Reply:
I have also updated these lines so that cs, vs, and ms aren't call calculated based off of clones